src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -954,6 +954,23 @@
       {:kind :invalid-signature
        :msg (tru "CSR contains a public key that does not correspond to the signing key")})))
 
+(schema/defn ensure-no-authorization-extensions!
+  "Throws an exception if the CSR contains authorization exceptions. These
+  extensions need to be signed using the puppet cert tool. This may change in
+  the future, but for now, it's too risky that certificates with authentication
+  extensions could be signed unintentionally."
+  [csr :- CertificateRequest]
+  (let [extensions (utils/get-extensions csr)]
+    (doseq [extension extensions]
+      (when (utils/subtree-of? ppAuthCertExt (:oid extension))
+        (sling/throw+
+         {:kind :disallowed-extension
+          :msg (format "%s %s"
+                (trs "CSR ''{0}'' contains an authorization extension, which is disallowed." csr)
+                (trs
+                 "Use `puppet cert --allow-authorization-extensions sign {0}` to sign this request."
+                 (get-csr-subject csr)))})))))
```

Comment:
This question mainly for my own understanding as I get used to this `trs` stuff ...

Is it necessary to use `format` and two `trs` calls here?  It seems to me like this could just be one `trs` call, but I could be misunderstanding something.